lib/engine/game/g_18_ka/game.rb

```diff
@@ -923,6 +942,69 @@ def setup
           national.add_ability(self.class::NATIONAL_FORCED_WITHHOLD_ABILITY)
         end
 
+        def route_distance_str(route)
+          towns = route.visited_stops.count(&:town?)
+          cities = route_distance(route) - towns
+          "#{cities}+#{towns}"
+        end
+
+        def revenue_for(route, stops)
```

Comment:
why do you need this?

Reply:
This is to reset the behavior of 1856's revenue_for, I didn't see an elegant way to call the base revenue_for so I re implemented it here